File: C#/PrimeNumberGeneratorTests/IsPrimeTests.cs
using System;
using System.Text;
using System.Collections.Generic;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using PrimeNumberGeneratorExercise.BusinessLogic;
using PrimeNumberGeneratorExercise.Interfaces;

namespace PrimeNumberGeneratorTests
{
    [TestClass]
    public class IsPrimeTests
    {
        IPrimeNumberGenerator _primeNumGenerator;

        [TestInitialize]
        public void Initialize()
        {
            _primeNumGenerator = new PrimeNumberGenerator();
        }

        [TestMethod]
        public void HappyPath_EvenPrimeNumber_Two()
        {
            bool isPrime = _primeNumGenerator.IsPrime(2);

            Assert.IsTrue(isPrime);
        }

        [TestMethod]
        public void HappyPath_OddPrimeNumber_Three()
        {
            bool isPrime = _primeNumGenerator.IsPrime(3);

            Assert.IsTrue(isPrime);
        }

        [TestMethod]
        public void HappyPath_PrimeNumber_23()
        {
            bool isPrime = _primeNumGenerator.IsPrime(23);

            Assert.IsTrue(isPrime);
        }

        [TestMethod]
        public void HappyPath_PrimeNumber_7901()
        {
            bool isPrime = _primeNumGenerator.IsPrime(7901);

            Assert.IsTrue(isPrime);
        }

        [TestMethod]
        public void Negative_EvenCompositeNumber_Four()
        {
            bool isPrime = _primeNumGenerator.IsPrime(4);

            Assert.IsFalse(isPrime);
        }

        [TestMethod]
        public void Negative_OddCompositeNumber_21()
        {
            bool isPrime = _primeNumGenerator.IsPrime(21);

            Assert.IsFalse(isPrime);
        }

        [TestMethod]
        public void Negative_OddCompositeNumber_121()
        {
            bool isPrime = _primeNumGenerator.IsPrime(121);

            Assert.IsFalse(isPrime);
        }

        [TestMethod]
        public void HappyPath_PrimeNumber_631()
        {
            bool isPrime = _primeNumGenerator.IsPrime(631);

            Assert.IsTrue(isPrime);
        }

        [TestMethod]
        public void Negative_EvenCompositeNumber_7900()
        {
            bool isPrime = _primeNumGenerator.IsPrime(7900);

            Assert.IsFalse(isPrime);
        }

        [TestMethod]
        public void HappyPath_PrimeNumber_7919()
        {
            bool isPrime = _primeNumGenerator.IsPrime(7919);

            Assert.IsTrue(isPrime);
        }

        [TestMethod]
        public void HappyPath_OddPrimeNumber_MaxIntValue()
        {
            bool isPrime = _primeNumGenerator.IsPrime(int.MaxValue);

            Assert.IsTrue(isPrime);
        }

        [TestMethod]
        public void Negative_NotPrime_NegativeNumber()
        {
            bool isPrime = _primeNumGenerator.IsPrime(-4);

            Assert.IsFalse(isPrime);
        }

        [TestMethod]
        public void Negative_NotPrime_Zero()
        {
            bool isPrime = _primeNumGenerator.IsPrime(0);

            Assert.IsFalse(isPrime);
        }

        [TestMethod]
        public void Negative_NotPrime_One()
        {
            bool isPrime = _primeNumGenerator.IsPrime(1);

            Assert.IsFalse(isPrime);
        }
    }
}

File: Java/PrimeNumberGeneratorExercise/test/com/png/implementations/tests/isPrimeTests.java
package com.png.implementations.tests;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.*;

import com.png.implementations.PrimeNumberGenerator;
import com.png.interfaces.*;

class isPrimeTests {
	IPrimeNumberGenerator primeNumberGenerator;
	
	@BeforeEach
	void init() {
		primeNumberGenerator = new PrimeNumberGenerator();
	}
	
	@Test
	void happyPath_EvenPrimeNumber_Two() {
		boolean isPrime = primeNumberGenerator.isPrime(2);
		
		assertTrue(isPrime);
	}

	@Test
	void happyPath_OddPrimeNumber_Three() {
		boolean isPrime = primeNumberGenerator.isPrime(3);
		
		assertTrue(isPrime);
	}
	
	@Test
	void happyPath_PrimeNumber_7901() {
		boolean isPrime = primeNumberGenerator.isPrime(7901);
		
		assertTrue(isPrime);
	}
	
	@Test
	void negative_EvenCompositeNumber_Four() {
		boolean isPrime = primeNumberGenerator.isPrime(4);
		
		assertFalse(isPrime);
	}
	
	@Test
	void negative_OddCompositeNumber_21() {
		boolean isPrime = primeNumberGenerator.isPrime(21);
		
		assertFalse(isPrime);
	}
	
	@Test
	void happyPath_PrimeNumber_23() {
		boolean isPrime = primeNumberGenerator.isPrime(23);
		
		assertTrue(isPrime);
	}
	
	@Test
	void negative_OddCompositeNumber_121() {
		boolean isPrime = primeNumberGenerator.isPrime(121);
		
		assertFalse(isPrime);
	}
	
	@Test
	void happyPath_PrimeNumber_631() {
		boolean isPrime = primeNumberGenerator.isPrime(631);
		
		assertTrue(isPrime);
	}
	
	@Test
	void negative_EvenCompositeNumber_7900() {
		boolean isPrime = primeNumberGenerator.isPrime(7900);
		
		assertFalse(isPrime);
	}
	
	@Test
	void happyPath_PrimeNumber_7919() {
		boolean isPrime = primeNumberGenerator.isPrime(7919);
		
		assertTrue(isPrime);
	}
	
	@Test
	void happyPath_OddPrimeNumber_MaxIntValue() {
		boolean isPrime = primeNumberGenerator.isPrime(Integer.MAX_VALUE);
		
		assertTrue(isPrime);
	}
	
	@Test
	void negative_NotPrime_NegativeNumber() {
		boolean isPrime = primeNumberGenerator.isPrime(-4);
		
		assertFalse(isPrime);
	}
	
	@Test
	void negative_NotPrime_Zero() {
		boolean isPrime = primeNumberGenerator.isPrime(0);
		
		assertFalse(isPrime);
	}
	
	@Test
	void negative_NotPrime_One() {
		boolean isPrime = primeNumberGenerator.isPrime(1);
		
		assertFalse(isPrime);
	}
}
